src/cosl/coordinated_workers/worker.py

```diff
@@ -170,14 +170,11 @@ def __init__(
             if self._resources_requests_getter
             else None
         )
+        # holistic update logic
+        self._holistic_update()
```

Comment:
"common exit hook" is fun too, but the established term for this is `reconcile`. Correct me if I'm wrong @ca-scribner?